src/Uno.UI.Runtime.Skia.Wpf/Input/WpfCorePointerInputSource.cs

```diff
@@ -230,21 +259,86 @@ private IntPtr OnWmMessage(IntPtr hwnd, int msg, IntPtr wparamOriginal, IntPtr l
 	#endregion
 
 	#region Convert helpers
-	private PointerEventArgs BuildPointerArgs(WpfMouseEventArgs args)
+	private PointerEventArgs BuildPointerArgs(InputEventArgs args)
 	{
 		if (args is null)
 		{
 			throw new ArgumentNullException(nameof(args));
 		}
 
-		var position = args.GetPosition(_hostControl);
-		var properties = BuildPointerProperties(args).SetUpdateKindFromPrevious(_previous?.CurrentPoint.Properties);
+		Point position;
+		PointerPointProperties properties;
+
+		uint pointerId;
+		if (args is WpfMouseEventArgs mouseEventArgs)
+		{
+			pointerId = 1;
+			position = mouseEventArgs.GetPosition(_hostControl);
+			properties = new()
+			{
+				IsLeftButtonPressed = mouseEventArgs.LeftButton == MouseButtonState.Pressed,
+				IsMiddleButtonPressed = mouseEventArgs.MiddleButton == MouseButtonState.Pressed,
+				IsRightButtonPressed = mouseEventArgs.RightButton == MouseButtonState.Pressed,
+				IsXButton1Pressed = mouseEventArgs.XButton1 == MouseButtonState.Pressed,
+				IsXButton2Pressed = mouseEventArgs.XButton2 == MouseButtonState.Pressed,
+				IsPrimary = true,
+				IsInRange = true
+			};
+		}
+		else if (args is StylusEventArgs stylusEventArgs)
+		{
+			pointerId = (uint)stylusEventArgs.StylusDevice.Id;
+			position = stylusEventArgs.GetPosition(_hostControl);
+
+			properties = new()
+			{
+				IsLeftButtonPressed = true,
+				IsPrimary = true,
+				IsInRange = !stylusEventArgs.InAir,
```

Comment:
This is actually always `true` except for some `Cancel` (and maybe also `Release` for touch).

Reply:
From what I remember of my investigations, the `IsInRange` is about "do we know where the pointer currently is?", no matter if it's pressed or not.

* For mouse: does the mouse is still connected on the computer (or the USB cable as been unplugged)
* For pen: does the pen is in the detectable range (over) or touching the screen (pressed)
* For touch: does a finger is touching the screen

So when we raise a pointer event it's almost always `true` (except some `Cancel` and `Release` as mentioned). BUT you can actually keep a reference of a pointer events args (or the [`Pointer`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.input.pointerroutedeventargs.pointer?view=windows-app-sdk-1.5)) for longer than the pointer event handler scope! Properties on that args/`Pointer` will then be updated by WinUI (this is currently not supported by uno BTW), and the `IsInRange` can become `false` in those cases.

(Note: not sure about the `IsInRange` state in windowed environment, when the pointer is above another window)